### PR TITLE
Add a copy button to spawn groups

### PR DESCRIPTION
**Pull request by TranquillyUnpleasant**
It does exactly what you expect it to:
image

Other changes:
- Made sorting use comps and have a secondary sorting parameter to prevent this weird group jumping when copying groups with similar sorting parameters:

https://user-images.githubusercontent.com/62061444/132419627-46db39e4-d32e-4dc3-95ec-85131ad04502.mp4

- `buildGroups()` is called instead of `updateWaves()` when removing a spawn group to prevent desyncs between the UI and the value of expanded group, otherwise you can run into situations where clicking a group doesn't expand it and you have to click again:

https://user-images.githubusercontent.com/62061444/132419773-f52a6e01-1df7-418d-8511-f8b4e7ef9609.mp4

(This was already an issue but it didn't matter before since identical groups were basically non-existant before)

If this is too many items for the bar i still have the other layout for copying groups i can make a pr for.

---
If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [x] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [x] I have ensured that my code compiles, if applicable.
- [x] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

File: core/src/mindustry/game/SpawnGroup.java [hunk 1]
```diff
@@ -157,6 +157,22 @@ public String toString(){
         '}';
     }
 
+    public SpawnGroup copy(){
```
Comment (Anuken) [inline core/src/mindustry/game/SpawnGroup.java:160]: Implement `Cloneable` and use `clone()` instead, it's faster and future-proof.

**Comment by TranquillyUnpleasant** [comment]
This is the correct way to implement this right? From what i can see you cant use `clone()` unless you override the method.

File: core/src/mindustry/game/SpawnGroup.java [hunk 2]
```diff
@@ -157,20 +157,9 @@ public String toString(){
...
-        newGroup.max = this.max;
-        newGroup.unitScaling = this.unitScaling;
-        newGroup.unitAmount = this.unitAmount;
-        newGroup.shields = this.shields;
-        newGroup.shieldScaling = this.shieldScaling;
-        newGroup.effect = this.effect;
-        newGroup.items = this.items;
-        return newGroup;
+    @Override
+    public SpawnGroup clone() throws CloneNotSupportedException{
```
Comment (Anuken) [inline core/src/mindustry/game/SpawnGroup.java:161]: I meant `copy()` should call `clone()` and handle the exception, see how `Weapon` does it.

**Comment by Barabara343434** [comment]
Güvenlikli

**Pull request merged**